### ISSUE TITLE
How does --threshold work?

### ISSUE THREAD

**Issue by off99555** [thtrieu/darkflow#587]
What is its default value?
How do I report all hundreds bounding boxes?
I tried setting `--threshold` to `0` and it doesn't show hundred bounding boxes like I expected.
What is the range that threshold can take on? Is it 0 to 1?

**Comment by troydeng** [comment]
yeah, it's between 0 to 1.
In the darkflow/net/yolo/__init__.py file, notice the last few lines:
```
def constructor(self, meta, FLAGS):
    def _to_color(indx, base):
        """ return (b, r, g) tuple"""
        base2 = base * base
        b = 2 - indx / base2
        r = 2 - (indx % base2) / base
        g = 2 - (indx % base2) % base
        return (b * 127, r * 127, g * 127)  ##TODO: ENTER

    if 'labels' not in meta:
        misc.labels(meta, FLAGS)  # We're not loading from a .pb so we do need to load the labels
    assert len(meta['labels']) == meta['classes'], (  # TODO: ENTER
            'labels.txt and {} indicate' + ' '
                                           'inconsistent class numbers'
    ).format(meta['model'])

    # assign a color for each label
    colors = list()
    base = int(np.ceil(pow(meta['classes'], 1. / 3)))
    for x in range(len(meta['labels'])):
        colors += [_to_color(x, base)]
    meta['colors'] = colors
    self.fetch = list()
    self.meta, self.FLAGS = meta, FLAGS

    #############################################
    #NOTICE THIS !
    # over-ride the threshold in meta if FLAGS has it.
    if FLAGS.threshold > 0.0:
        self.meta['thresh'] = FLAGS.threshold```

If you set it to 0.00001, you'll see hundreds of boxes

**Comment by off99555** [comment]
I see. But I've tried 0.0001 before and it didn't work. Maybe I'll need to try again.

**Comment by troydeng** [comment]
I guess 'cause there's a non-maximum suppression, setting it to 0.0001 is not so obvious as 0.00001, haha

**Comment by off99555** [comment]
OK. I'll try even smaller number. lol

**Comment by mohamedabdallah1996** [comment]
I tried threshold 0.001, my loss is ~1.6 but it has very bad accuracy.
I can't reduce the loss much more  !!
what can I do?

**Comment by off99555** [comment]
I don't think the threshold would effect training loss, or does it?

**Comment by off99555** [comment]
Usually it's a bug of the training process that we cannot see any bounding boxes. I can't remember what I did that made it possible to see the boxes. But it is possible and it requires retraining.
